### ISSUE TITLE
[clang] 18.1.6-final: FAILED: lib/Target/PowerPC/CMakeFiles/LLVMPowerPCCodeGen.dir/PPCRegisterInfo.cpp.o

### ISSUE THREAD

**Issue by azhan92** [llvm/llvm-project#93304]
```
FAILED: lib/Target/PowerPC/CMakeFiles/LLVMPowerPCCodeGen.dir/PPCRegisterInfo.cpp.o 
/scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/Phase1/Release/llvmCore-18.1.6-final.install/usr/local/bin/clang++ -DGTEST_HAS_RTTI=0 -D_LARGE_FILE_API -D_XOPEN_SOURCE=700 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/Phase2/Release/llvmCore-18.1.6-final.obj/lib/Target/PowerPC -I/scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/llvm-project/llvm/lib/Target/PowerPC -I/scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/Phase2/Release/llvmCore-18.1.6-final.obj/include -I/scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/llvm-project/llvm/include "-pthread" -mcmodel=large -fPIC -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG -std=c++17 -fPIC  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Target/PowerPC/CMakeFiles/LLVMPowerPCCodeGen.dir/PPCRegisterInfo.cpp.o -MF lib/Target/PowerPC/CMakeFiles/LLVMPowerPCCodeGen.dir/PPCRegisterInfo.cpp.o.d -o lib/Target/PowerPC/CMakeFiles/LLVMPowerPCCodeGen.dir/PPCRegisterInfo.cpp.o -c /scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/llvm-project/llvm/lib/Target/PowerPC/PPCRegisterInfo.cpp
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.      Program arguments: /scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/Phase1/Release/llvmCore-18.1.6-final.install/usr/local/bin/clang++ -DGTEST_HAS_RTTI=0 -D_LARGE_FILE_API -D_XOPEN_SOURCE=700 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/Phase2/Release/llvmCore-18.1.6-final.obj/lib/Target/PowerPC -I/scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/llvm-project/llvm/lib/Target/PowerPC -I/scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/Phase2/Release/llvmCore-18.1.6-final.obj/include -I/scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/llvm-project/llvm/include -pthread -mcmodel=large -fPIC -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG -std=c++17 -fPIC -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Target/PowerPC/CMakeFiles/LLVMPowerPCCodeGen.dir/PPCRegisterInfo.cpp.o -MF lib/Target/PowerPC/CMakeFiles/LLVMPowerPCCodeGen.dir/PPCRegisterInfo.cpp.o.d -o lib/Target/PowerPC/CMakeFiles/LLVMPowerPCCodeGen.dir/PPCRegisterInfo.cpp.o -c /scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/llvm-project/llvm/lib/Target/PowerPC/PPCRegisterInfo.cpp
1.      <eof> parser at end of file
2.      Code generation
3.      Running pass 'PPC Merge String Pool' on module '/scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/llvm-project/llvm/lib/Target/PowerPC/PPCRegisterInfo.cpp'.
clang++: error: clang frontend command failed with exit code 139 (use -v to see invocation)
clang version 18.1.6 (https://github.com/llvm/llvm-project.git 1118c2e05e67a36ed8ca250524525cdb66a55256)
Target: powerpc64-ibm-aix7.2.0.0
Thread model: posix
InstalledDir: /scratch/jenkins/releasetest/workspace/LLVM-Release/release-testing-prototype/llvm/clang18.1.6-final/final/Phase1/Release/llvmCore-18.1.6-final.install/usr/local/bin
```
[PPCRegisterInfo.tar.gz](https://github.com/llvm/llvm-project/files/15435720/PPCRegisterInfo.tar.gz)

This [commit](https://github.com/llvm/llvm-project/commit/1184a9cb30e6a12c883b918867f2f06bc3096fc0) seems to be the culprit.

**Comment by nikic** [comment]
Looks like the same issue as https://github.com/llvm/llvm-project/issues/92991, with a fix at https://github.com/llvm/llvm-project/pull/92996.

**Comment by chenzheng1030** [comment]
https://github.com/llvm/llvm-project/pull/93442 is posted to cherry-pick the fix to 18.x. Waiting for merge and verify

**Comment by chenzheng1030** [comment]
@azhan92 This bug should be fixed in latest 18.1.7 release? Thanks.

**Comment by azhan92** [comment]
@chenzheng1030 Yes, I no longer see it in the current release, thank you
